app/models/manageiq/providers/redhat/infra_manager.rb

```diff
@@ -103,24 +103,39 @@ def vm_reconfigure(vm, options = {})
   end
 
   def add_disks(add_disks_spec, vm)
-    ems_storage_uid = add_disks_spec["ems_storage_uid"]
+    storage = add_disks_spec["storage"]
     with_disk_attachments_service(vm) do |service|
-      add_disks_spec["disks"].each { |disk_spec| service.add(prepare_disk(disk_spec, ems_storage_uid)) }
+      add_disks_spec["disks"].each { |disk_spec| service.add(prepare_disk(disk_spec, storage)) }
     end
   end
 
-  def prepare_disk(disk_spec, ems_storage_uid)
-    {
-      :bootable  => disk_spec["bootable"],
-      :interface => "VIRTIO",
-      :active    => true,
-      :disk      => {
-        :provisioned_size => disk_spec["disk_size_in_mb"].to_i * 1024 * 1024,
-        :sparse           => disk_spec["thin_provisioned"],
-        :format           => disk_spec["format"],
-        :storage_domains  => [:id => ems_storage_uid]
-      }
-    }
+  # prepare disk attachment request payload of adding disk for reconfigure vm
+  def prepare_disk(disk_spec, storage)
+    disk_attachment_builder = DiskAttachmentBuilder.new
```

Comment:
done - you're right. what's the point on having options in c'tor if not using them...